packages/nuxt/src/imports/module.ts

```diff
@@ -119,22 +131,20 @@ export default defineNuxtModule<Partial<ImportsOptions>>({
     })
 
     // Watch composables/ directory
-    const templates = [
-      'types/imports.d.ts',
-      'imports.d.ts',
-      'imports.mjs'
-    ]
     nuxt.hook('builder:watch', async (_, relativePath) => {
       const path = resolve(nuxt.options.srcDir, relativePath)
       if (composablesDirs.some(dir => dir === path || path.startsWith(dir + '/'))) {
-        await updateTemplates({
-          filter: template => templates.includes(template.filename)
-        })
+        await regenerateImports()
       }
     })
 
-    nuxt.hook('app:templatesGenerated', async () => {
-      await regenerateImports()
```

Comment:
Previously we did regeneration (rescan) after all templates to avoid race conditions in #21934. But since unimport also use templates to generate those dts etc, so it means we missed the change to update them - until this get triggered - we are always one "tick" behind.

This PR makes sure `regenerateImports` always update (only) the unimport templates. Thus we also need some mechanism to dedupe and avoid circular update